src/scripts/eastern_kingdoms/redridge_mountains/redridge_mountains.cpp

```diff
@@ -63,6 +63,29 @@ struct npc_corporal_keeshan_escortAI : npc_escortAI
         m_uiShieldBashTimer  = 8000;
     }
 
+    void JustRespawned() override
+    {
+        // Reset EscortAI
+        npc_escortAI::JustRespawned();
+        Reset(); 
+        float m_homeX, m_homeY, m_homeZ, m_homeO;
+
+        //Reset Faction and Quest Giver Flag
+        m_creature->SetFactionTemplateId(FACTION_STORMWIND);
+        m_creature->SetFlag(UNIT_NPC_FLAGS, UNIT_NPC_FLAG_QUESTGIVER);
+
+        // SetSpawn
+        if (m_homeX || m_homeY || m_homeZ)
```

Comment:
m_homeX etc are never set - this if statement is unpredictable.

In debug the variables will be initialized to 0. But on release build the values will be uninitialized.

the m_ prefix should be used for member variables only.